### PR TITLE
Add a stdin onramp.

### LINKED ISSUE #509: Additional Linked sources / sinks
There are some additional sources that could take some linked transport goodness:

* [ ] stdin/stdout/stderr source
* [ ] tcp source
* [ ] tcp sink

## stdin/stdout/stderr source

read from stdin write to stdout, errors go to stderr

## tcp source

stream replie to requests over the tcp connection

## tcp sink 

stream replies for requests on a tcp connection

### PR DESCRIPTION
**Pull request by aryan9600**
Signed-off-by: Sanskar Jaiswal <jaiswalsanskar078@gmail.com>

# Pull request

## Description
Add an onramp which takes inputs from stdin.

## Related

* Related Issues: fixes #509 
* Related [docs PR](https://github.com/tremor-rs/tremor-www-docs/pull/99)

## Checklist

* [x] The RFC, if required, has been submitted and approved
* [x] Any user-facing impact of the changes is reflected in docs.tremor.rs
* [x] The code is tested
* [x] Use of unsafe code is reasoned about in a comment
* [x] Update CHANGELOG.md appropriately, recording any changes, bug fixes, or other observable changes in behavior
* [x] The performance impact of the change is measured (see below)

### CODE REVIEW

File: src/source/stdin.rs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+#[async_trait::async_trait()]
+impl Source for Int {
+    fn id(&self) -> &TremorUrl {
+        &self.onramp_id
+    }
+
+    async fn pull_event(&mut self, _id: u64) -> Result<SourceReply> {
+        let mut input = String::new();
+        if let Ok(len) = self.stdin.read_line(&mut input).await {
```
Comment (Licenser) [inline src/source/stdin.rs:81]: We shouldn't split the input into lines here, that's the job of the `lines` preprocessor. That way it mirrors the behaviour of the `stdout` sink where the `lines` postprocessor adds the newlines to the output (if required)

**Review by Licenser — CHANGES_REQUESTED** [review]
👍 nice job, only small thing is not to use `read_line` but just `read` otherwise looks great :D

File: src/source/stdin.rs (hunk 1 above)
Comment (aryan9600) [reply src/source/stdin.rs:81]: I don't think `read_line()` splits the input into lines. For example: passing an input like `snot \n badger`, would be treated as one single string "snot \n badger", which is similar to how the websocket onramp works for example. Do let me know if I'm missing something here :D

Comment (Licenser) [reply src/source/stdin.rs:81]: The [documentation](https://docs.rs/async-std/0.99.7/async_std/io/struct.Stdin.html#method.read_line) says "Reads a line of input" so I'm fairly sure it does read a line.

The second issue, enforcing UTF8 compliance as it uses a `String` also is relevant.

Comment (aryan9600) [reply src/source/stdin.rs]: Done! ✅ @Licenser

Comment (Licenser) [reply src/source/stdin.rs]: nice :D thank you!

**Review by Licenser — APPROVED** [review]
Thank you! This is great :D

**Pull request merged**